### DISCUSSION TITLE
On Linux desktops, alternate WMs, and a different path

### DISCUSSION THREAD

**Discussion by cerebrate** [microsoft/wslg#516] [Ideas]
Just a quick shout-out to everyone interested in running a full Linux desktop, alternate window manager, or suchlike using WSLg that there is potentially the option of using one of the various nested X server products (Xephyr, Xnest, etc.) to do so in a window; potentially a "full-screen" window, getting the desktop you want via a somewhat roundabout route.

As of the current WSL/WSLg versions, Xnest appears to work just fine;

`Xnest :1` creates exactly the X-in-a-window display you're looking for.

(It's less capable for various reasons than Xephyr, which still freezes shortly after startup but I'm still hoping we can get working under WSLg - my issue here: https://github.com/microsoft/wslg/issues/154 - but it may well satisfy your needs when properly tweaked.)

**Comment by mwoodpatrick** [comment]
I'm interested in doing this possibly from an os running in a container of VM running under wsl-2. I'd be interested in anyone's experience doing this.
